### PR TITLE
Refactor `cont.wast` to not use `let` & `func.bind`

### PR DESCRIPTION
**Pull request by dhil**
This patch refactors the test `cont.wast` such that it no longer uses the obsolete `let` instruction.

### CODE REVIEW

**Comment by dhil** [comment]
@rossberg @slindley PTAL. This patch brings us closer to be able to integrate `cont.wast` into wasm-tools/wasmtime testsuite. ~~We still need to rewrite the parts that use `func.bind`~~.

File: test/core/cont.wast [hunk 1]
```diff
@@ -86,18 +86,18 @@
     (unreachable)
   )
   (func $nl3 (param $k (ref $k1))
+    (local $k' (ref null $k1))
```
Comment (rossberg) [inline test/core/cont.wast:89]: This should work as well.
```suggestion
    (local $k' (ref $k1))
```

File: test/core/cont.wast [hunk 2]
```diff
@@ -214,18 +214,20 @@
 
   (func (export "sum") (param $i i64) (param $j i64) (result i64)
     (local $sum i64)
+    (local $n i64)
+    (local $k (ref null $cont))
```
Comment (rossberg) [inline test/core/cont.wast:218]:
```suggestion
    (local $k (ref $cont))
```

File: test/core/cont.wast (hunk 1 above)
Comment (dhil) [reply test/core/cont.wast:89]: I get the following error then
```
../test/core/cont.wast:89.16-89.25: invalid module: non-defaultable local type
```

**Comment by rossberg** [comment]
Oh, I suppose this repo hasn't been updated to the final locals semantics yet.

Would you be willing to merge from upstream function-references/main? I suspect there might be some conflicts, however...

**Comment by dhil** [comment]
> Oh, I suppose this repo hasn't been updated to the final locals semantics yet.
> 
> Would you be willing to merge from upstream function-references/main? I suspect there might be some conflicts, however...

Yes. I can try to sync in changes from function-references and exception-handling. But I would like to do this in a separate PR.

My latest commit further refactors `cont.wast` such that it no longer uses `func.bind`. With these changes I believe I can integrate it into the wasmtime testsuite.

**Review by slindley — APPROVED** [review]
Looks fine to me

**Pull request merged**